ansible/roles/python/vars/debian-12.yml

```diff
@@ -0,0 +1,22 @@
+---
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+python_packages:
+- python3
+- python3-pip
+- python3-setuptools
+- python3-virtualenv
+
+extra_pip_args: --break-system-packages
```

Comment:
@nick-stroud you'll see here that we may need to consider creating a virtual environment for Slurm going forward

Reply:
Can you note here what packages (or step) required this flag?

I am wondering 1) do we actually expect that we have somehow broken the system or is our case probably safe. 2) is it possible the suspect step is not needed, like upgrading pip.

Reply:
Debian 12 enforces a ban against system-wide installation of packages. The upgrade of pip is only one example of several where this solution does that. I'll add more clarifying context.

I agree that we may not need to upgrade pip here. I'd prefer to upgrade to a minimum version rather than blindly upgrade to latest but don't want to touch the implementation and scope in testing our supported OSes.